libp2p/peerstore.nim

```diff
@@ -107,36 +129,206 @@ proc set*(addressBook: var AddressBook,
   ## Consider using addressBook.add() as alternative.
   
   addressBook.book[peerId] = addrs
-  addressBook.addrChange(peerId, addressBook.get(peerId)) # Notify clients
+
+  # Notify clients
+  for handler in addressBook.changeHandlers:
+    handler(peerId, addressBook.get(peerId))
+
+#####################
+# Protocol Book API #
+#####################
+
+proc get*(protoBook: ProtoBook,
+          peerId: PeerID): HashSet[string] =
+  ## Get the known protocols of a provided peer.
+  
+  protoBook.book.getOrDefault(peerId,
+                              initHashSet[string]())
+
+proc add*(protoBook: var ProtoBook,
```

Comment:
At what point will this be called? Who is responsible for calling it and at what point in the lifecycle?

Reply:
Since a Peer Store provides a "standalone" API it can theoretically be instantiated and added to (at any time) by any module seeking to provide some level of peer management.
Eventually I envision it to be integrated in such a way that there is a single libp2p instance, with the following integration points (based on [js-implementation](https://github.com/libp2p/js-libp2p/tree/master/src/peer-store#submitting-records-to-the-peerstore)):
- a `Dialer` (TBD) adds "bootstrap" or statically defined peers when these are dialed. This is most likely the first point of integration for `WakuRelay` as well. The `Dialer` would also be responsible for maintaining connections and redialing peers when dropped.
- the `identify` protocol adds information about identified peers.
- Discovery/DHT mechanisms adds information about discovered peers.